src/generators/net_caffe_recurrent.cc

```diff
@@ -0,0 +1,371 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2014-2016 Emmanuel Benazera
+ * Author: Emmanuel Benazera <beniz@droidnik.fr>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "net_caffe_recurrent.h"
+#include "imginputfileconn.h"
+
+namespace dd
+{
+
+
+  void NetLayersCaffeRecurrent::add_basic_block(caffe::NetParameter *net_param,
+                                                const std::string &bottom_seq,
+                                                const std::string &bottom_cont,
+                                                const std::string &top,
+                                                const int &num_output,
+                                                const double &dropout_ratio,
+                                                const std::string &type,
+                                                const int id)
+  {
+    std::string ctype = (type == "R" ?  "RNN" : "LSTM");
+    caffe::LayerParameter *lparam = net_param->add_layer();
+    lparam->set_type(ctype);
+    lparam->set_name(ctype+std::to_string(id));
+    if (dropout_ratio ==0)
+      lparam->add_top(top);
+    else
+      lparam->add_top(ctype+std::to_string(id)+"_undropped");
+    lparam->add_bottom(bottom_seq);
+    lparam->add_bottom(bottom_cont);
+    caffe::RecurrentParameter *rparam = lparam->mutable_recurrent_param();
+    rparam->set_num_output(num_output);
+    caffe::FillerParameter *weight_filler_param = rparam->mutable_weight_filler();
+    weight_filler_param->set_type("uniform");
```

Comment:
OK, that's not used very often these days, is that fine initialization for recurrent layers ?

Reply:
it is the one used in pytorch ...

Reply:
OK, for OCR we use Xavier, can we fill the value with a parameter to cover other values ?

Reply:
until which level should the parameter be exposed ? it is now in add_basic_block, but not in configure_net (having it in configure_net means exposing it at API level)